experiments/dgps.R

```diff
@@ -0,0 +1,125 @@
+# dgps.R - convenience script for generating simulation data for grf.
+# Usage:
+# `source("dgps.R")` exports:
+# gen_data() (documented below)
+
+
+#' Generate comparable simulation data
+#'
+#' The following function returns one simulated data set from one of the seven following DGPs:
+#' "simple", "aw1", "aw2", "aw3", "ai1", "ai2", "kunzel".
+#'
+#' Each DGP is parameterized by
+#' x: observables
+#' m: conditional mean of Y
+#' tau: treatment effect
+#' e: propensity scores
+#' V: conditional variance of Y
+#'
+#' The following rescaled data is returned
+#' m = m / sd(m) * sigma.m
+#' tau = tau / sd(tau)* sigma.tau
+#' V = V / mean(V) * sigma.noise^2
+#' W = rbinom(e)
+#' Y = m + (W - e) * tau + sqrt(V) + rnorm(n)
+#'
+#' @param n The number of observations
+#' @param p The number of covariates (note: the minium varies by DGP)
+#' @param sigma.m The standard deviation of the unconditional mean of Y. Default=1.
+#' @param sigma.tau The standard deviation of the treatment effect. Default=0.1.
+#' @param sigma.noise The conditional variance of Y. Default=1.
+#' @param dgp The kind of dgp. Default="simple".
+#'
+#' @return A list consisting of:
+#'  X, Y, W, tau, m, e, dgp.
+#'
+#' @examples
+#' gen_data(100, 5, sigma.m = 1, sigma.tau = 0.1, sigma.noise = 1, dgp = "simple")
+#' @note:
+#' To add an additonal DGP, fill in the template below and add an entry to `dgp` and `.minp`.
+#'
+gen_data <- function(n, p, sigma.m = 1, sigma.tau = 0.1, sigma.noise = 1,
+                     dgp = c("simple", "aw1", "aw2", "aw3", "ai1", "ai2", "kunzel"),
+                     ...) {
+  .minp <- c(3, 2, 2, 1, 2, 6, 2)
```

Comment:
It'd be more robust to store this as a named list: c(simple=3, ...)

Then, you can access it as minp = .minp[dgp]